### DISCUSSION TITLE
Add list of most popular packages

### DISCUSSION THREAD

**Discussion by tiwoc** [SwiftPackageIndex/SwiftPackageIndex-Server#1348] [Ideas]
I'd love to see a list of the most popular packages. They could be ranked by GitHub stars and/or by the number of other packages that depend on them.

This would allow developers to discover popular packages that they're either not aware of at all, or that they've heard of but didn't know they were widely used.

**Comment by daveverwer** [comment]
Thank you for your feedback and suggestion!

I think there's a great case for this. I picked Recently Added and Recent Releases as they're more likely to change often, and I think a lot of the really big packages are already well known. I would like to add much more "list" type content though, and top packages will almost certainly be part of that.

**Comment by daveverwer** [comment]
So this was mentioned recently in #769 and I just wrote my current thoughts on popularity that I was going to post over there, but here is a better place for it.

Popularity is a weird one. Certainly, the number of stars is an indicator, and having a list of popular packages is potentially interesting, but it’s not something I’m rushing to prioritise for a few reasons:

1. As I mentioned above, it won’t change much, and people already know what the popular packages are because they get a lot of coverage. Without looking, I’d bet that Alamofire and AFNetworking lead it and that SDWebImage, RxSwift, and ReactiveCocoa will all be high up in the list. That list hasn’t changed significantly in years, and I don’t think it’s that interesting to browse.
2. Popularity without categories is fairly meaningless. Most popular networking library is much more useful than the most popular overall package. I’d be more in favour of prioritising a this when we have categories implemented with significant data in them.
3. Popularity isn’t how people should be picking dependencies. Certainly popularity is an indicator of quality as people are unlikely to star a package they think is poor quality, but it’s not more than one (of many) indicators, and I don’t think we should rely too much on it without the other indicators. I think we should be wary of drawing this metric out on its own too much.

I also don’t really like the idea of using dependencies as a measure of popularity. Dependency information can be a good indicator, but not all packages can be dependencies, so it skews the list towards smaller, µpackages.
